yew/src/html/listener/listener_web_sys.rs

```diff
@@ -1,100 +1,273 @@
 // Inspired by: http://package.elm-lang.org/packages/elm-lang/html/2.0.0/Html-Events
+
+use wasm_bindgen::JsCast;
+use web_sys::{Element, Event, HtmlInputElement, HtmlSelectElement, HtmlTextAreaElement};
+
+macro_rules! impl_action {
+    ($($action:ident($type:ident) -> $ret:path => $convert:path)*) => {$(
+        impl_action!($action($type, ::std::u8::MIN) -> $ret => $convert);
+    )*};
+    ($($action:ident($type:ident, $flags:path) -> $ret:path => $convert:path)*) => {$(
+        /// An abstract implementation of a listener.
+        #[doc(hidden)]
+        pub mod $action {
+            use crate::callback::Callback;
+            use crate::virtual_dom::Listener;
```

Comment:
Why not just import these at the top?

Reply:
Does not seem to work because of the module boundaries between `listener_web_sys` and the created child modules.

Reply:
Right, I forgot that we're creating a module for each type. Apologies for that